src/idl/src/processor.c

```diff
@@ -380,7 +411,17 @@ idl_retcode_t idl_parse(idl_pstate_t *pstate)
     }
   }
 
-  return ret;
+  /*check that the extensibilities of inherited and inheriting match*/
+  idl_visitor_t visitor;
+  memset(&visitor, 0, sizeof(visitor));
+
+  visitor.visit = IDL_STRUCT;
+  visitor.accept[IDL_ACCEPT_STRUCT] = &check_inheritance;
+
+  if (ret == IDL_RETCODE_OK)
+    return idl_visit(pstate, pstate->root, &visitor, NULL);
```

Comment:
While this technically works. The visitor pattern is meant for consumers. This works, but is slower because it traverses much more than just structs. Also, I'd like to keep the split intact.